spec/services/users/identity/unlink_service_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Users::Identity::UnlinkService do
+  subject(:service_response) { service.execute }
+
+  let(:service) do
+    described_class.new(current_user, identity)
+  end
+
+  context 'when user is valid' do
+    let(:provider_id) { :google }
+    let(:current_user) { create(:user) }
+    let(:identity) { create(:user_identity, user: current_user, identifier: 'identifier', provider_id: :google) }
```

Comment:
To make the test pass, the identity needs to be created at the beginning of the spec and not during the subject.
```suggestion
    let!(:identity) { create(:user_identity, user: current_user, identifier: 'identifier', provider_id: :google) }
```